src/streaming-client.ts

```diff
@@ -95,6 +95,7 @@ export class RevAiStreamingClient extends EventEmitter {
      * Signals to the api that you have finished sending data.
      */
     public end(): void {
+        this.requests.emit('ending');
```

Comment:
This event allows us to sync the ending of the request stream with the ending of the overlying duplex stream. Without this there is a period of time where the requests stream is draining the final input message and is not writable but has not yet emitted the 'finish' event which we would use to notify the duplex that it should finish as well. This way we signal the duplex to set writable = false and stop accepting new input while the underlying request stream finishes inserting the last EOS message